plugins/backstage-plugin-coder/docs/guides/coder-api.md

```diff
@@ -0,0 +1,257 @@
+- `useCoderAuth` - Provides methods and state values for interacting with your current Coder auth session from within Backstage.
+
+#### Convenience hooks
+
+- `useCoderQuery` - Simplifies wiring up `useQuery`, `useCoderApi`, and `useCoderAuth`
+
+## Recommendations for accessing the API
+
+1. If querying data, prefer `useCoderQuery`. It automatically wires up all auth logic to React Query (which includes pausing queries if the user is not authenticated). It also lets you access the Coder API via its query function. `useQuery` is also a good escape hatch if `useCoderQuery` doesn't meet your needs, but it requires more work to wire up correctly.
+2. If mutating data, you will need to call `useMutation`, `useQueryClient`, and `useCoderApi` in tandem\*. The plugin exposes a `CODER_QUERY_KEY_PREFIX` constant that you can use to group all Coder queries together.
+
+We highly recommend **not** fetching with `useState` + `useEffect`, or with `useAsync`. Both face performance issues when trying to share state. See [ui.dev](https://www.ui.dev/)'s wonderful [_The Story of React Query_ video](https://www.youtube.com/watch?v=OrliU0e09io) for more info on some of the problems they face.
+
+\* A `useCoderMutation` hook is in the works to simplify wiring these up.
+
+### Comparing query caching strategies
+
+|                                                                    | `useState` + `useEffect` | `useAsync` | `useQuery` | `useCoderQuery` |
+| ------------------------------------------------------------------ | ------------------------ | ---------- | ---------- | --------------- |
+| Automatically handles race conditions                              | 🚫                       | ✅         | ✅         | ✅              |
+| Can retain state after component unmounts                          | 🚫                       | 🚫         | ✅         | ✅              |
+| Easy, on-command query invalidation                                | 🚫                       | 🚫         | ✅         | ✅              |
+| Automatic retry logic when a query fails                           | 🚫                       | 🚫         | ✅         | ✅              |
+| Less need to fight dependency arrays                               | 🚫                       | 🚫         | ✅         | ✅              |
+| Easy to share state for sibling components                         | 🚫                       | 🚫         | ✅         | ✅              |
+| Pre-wired to Coder auth logic                                      | 🚫                       | 🚫         | 🚫         | ✅              |
+| Can consume Coder API directly from query function                 | 🚫                       | 🚫         | 🚫         | ✅              |
+| Automatically groups Coder-related queries by prefixing query keys | 🚫                       | 🚫         | 🚫         | ✅              |
+
```

Comment:
In line with my comment above, I don't know that we need the `useState` + `useEffect` column.